deployer/deploy.sh

```diff
@@ -22,9 +22,16 @@ osds=$((${QUOTA_COUNT:-1} * 3))
 
 echo "Quota count: ${QUOTA_COUNT} -- OSD count: ${osds}"
 
+#-- The path mapped to StorageCluster configmap used for overriding.
+overridePath="/sc-override/storagecluster.yml"
+
 #-- Patch the OSD count into the StorageCluster & apply it
+#-- If config map containing StorageCluster exists, apply it. 
+#-- Else, apply the default StorageCluster.
 while true; do
-    sed "s/STORAGE_NODES/${osds}/" storagecluster.yml | \
+    storageClusterPath="storagecluster.yml"
```

Comment:
Does this need to be done inside the loop?

Reply:
Yes, it should fallback to the default StorageCluster if the config map is removed

Reply:
This could lead to issues - what if the OSD size is changed in the config map , the cluster is created and then the config map is deleted?

Reply:
The ability to use a config map to override the `storagecluster` is not meant to be available to anybody but the SREs who are supporting OCS on OSD, as such this is a crude tool that does not cover all cases.
It will be used with extreme caution and only after consultation.

Another point here is that this is just a temporary solution and a more wholesome solution, in the form of an operator based on operator-sdk, is coming in the near future.

@nbalacha I will just like to point out that the same issue can be reproduced by directly editing a `sotragecluster` CR, so the risk is already present on every OCS installation.

Reply:
That is true but deleting a ConfigMap seems a lot easier than editing the file, especially when there is a UI in the OCP console to prevent it.

Reply:
In principle, I agree but please take into account that the barrier will not be deleting the config map but getting admin access to the `openshift-sotrage` namespace in the first place.
In an OSD environment, there is literally no admin access to any openshift namespace.